### PR TITLE
Budget  App

### PR DESCRIPTION
**Pull request by Kwaleyela-Ikafa**
Changes made:

- Added authorization using cancancan
- Added Authentication using devise gem
- Fixed N+1 problems using bullet gem
- Created the page to add a category.
- Created the home page.
- Created the page to add a transaction.
- Created the transactions page.
- Added styling using bootstrap

[demo video](https://www.loom.com/share/2b8d337b71684612b17700a992b72ce3)

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      <% end %><br />
+      <%= f.password_field :password, autocomplete: "new-password" %>
+    </div>
+  
+    <div class="field">
+      <%= f.label :password_confirmation %><br />
+      <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+    </div>
+  
+    <div class="actions">
```
Comment (BertrandConxy) [inline app/views/devise/registrations/new.html.erb]: - [ ] Good job so far 👏 👏 However, I would suggest that you improve your design on the registration page. I have made the comparison below 👇 

**Given design**
image

**Your design**
image

Please improve it and make sure that you stick to the design given 😃 👍

File: app/views/devise/confirmations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email", value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email) %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Resend confirmation instructions" %>
+  </div>
+<% end %>
+
```
Comment (BertrandConxy) [inline app/views/devise/confirmations/new.html.erb:15]: - [ ] Please improve the login page design in order to match the design given 😃 👍 See the comparison below 👇 
**Given design**
image

**Your design**
image

Please make sure you stick to the design given 👍

File: README.md [hunk 3]
```diff
@@ -79,7 +79,7 @@ In order to run this project you need:
 ### Setup
 
 Clone this repository to your desired folder:
-- Clone [this repository](https://github.com/Kwaleyela-Ikafa/blog-app.git)
+- Clone [this repository](https://github.com/Kwaleyela-Ikafa/budget-app.git)
```
Comment (BertrandConxy) [inline README.md:82]: - [ ] Please give credit to the author of the original design in the `Acknowledgement` section as it is in the project section

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @Kwaleyela-Ikafa, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 
- No linter errors
- correct flow is used
- good PR


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**